internal/graph/dispatch_throttling_check_resolver.go

```diff
@@ -27,37 +26,44 @@ type DispatchThrottlingCheckResolverConfig struct {
 // in the throttling queue. One item form the throttling queue will be processed ticker.
 // This allows a check / list objects request to be gradually throttled.
 type DispatchThrottlingCheckResolver struct {
-	delegate        CheckResolver
-	config          DispatchThrottlingCheckResolverConfig
-	ticker          *time.Ticker
-	throttlingQueue chan struct{}
-	done            chan struct{}
+	delegate  CheckResolver
+	config    *DispatchThrottlingCheckResolverConfig
+	throttler throttler.Throttler
 }
 
 var _ CheckResolver = (*DispatchThrottlingCheckResolver)(nil)
 
-var (
-	dispatchThrottlingResolverDelayMsHistogram = promauto.NewHistogramVec(prometheus.HistogramOpts{
-		Namespace:                       build.ProjectName,
-		Name:                            "dispatch_throttling_resolver_delay_ms",
-		Help:                            "Time spent waiting for dispatch throttling resolver",
-		Buckets:                         []float64{1, 3, 5, 10, 25, 50, 100, 1000, 5000}, // Milliseconds. Upper bound is config.UpstreamTimeout.
-		NativeHistogramBucketFactor:     1.1,
-		NativeHistogramMaxBucketNumber:  100,
-		NativeHistogramMinResetDuration: time.Hour,
-	}, []string{"grpc_service", "grpc_method"})
-)
+// DispatchThrottlingCheckResolverOpt defines an option that can be used to change the behavior of DispatchThrottlingCheckResolver
+// instance.
+type DispatchThrottlingCheckResolverOpt func(checkResolver *DispatchThrottlingCheckResolver)
+
+// WithDispatchThrottlingCheckResolverConfig sets the config to be used for DispatchThrottlingCheckResolver.
+func WithDispatchThrottlingCheckResolverConfig(config DispatchThrottlingCheckResolverConfig) DispatchThrottlingCheckResolverOpt {
+	return func(r *DispatchThrottlingCheckResolver) {
+		r.config = &config
+	}
+}
+
+// WithThrottler sets the throttler to be used for DispatchThrottlingCheckResolver.
+func WithThrottler(throttler throttler.Throttler) DispatchThrottlingCheckResolverOpt {
+	return func(r *DispatchThrottlingCheckResolver) {
+		r.throttler = throttler
+	}
+}
 
-func NewDispatchThrottlingCheckResolver(
-	config DispatchThrottlingCheckResolverConfig) *DispatchThrottlingCheckResolver {
+func NewDispatchThrottlingCheckResolver(opts ...DispatchThrottlingCheckResolverOpt) *DispatchThrottlingCheckResolver {
```

Comment:
Thought: Right now all the check resolvers are instantiated at the server level, but for this new resolver there is nothing preventing it from being instantiated at the request level, because in reality the shared state is only the `throttler` field.

i'm worried for the future, when we add options to tune caching per-request, the cached Check Resolver will have to be instantiated at the request level, this will throw off the entire resolver onion?